### PR TITLE
[MIG][11.0] cms_form

### PR DESCRIPTION
**Pull request by simahawk**

### CODE REVIEW

**Comment by tschanzt** [comment]
Depends on #35. cms_status_message is a dependency.

**Review by tschanzt — APPROVED** [review]
Looks good to me.

**Comment by simahawk** [comment]
tests currently broken because of this https://github.com/odoo/odoo/pull/22384
Need to fix the tmp oca dependency.

**Comment by simahawk** [comment]
Included https://github.com/OCA/website-cms/pull/36/commits/0f7715dfca77cf2239f0798955ffe48837deac56 to fix odoo till upstream PR is merged.

**Comment by pedrobaeza** [comment]
I'm wondering is this overlaps with the new website_form_builder. Can you tell me, Simone?

**Comment by pedrobaeza** [comment]
https://github.com/OCA/website/tree/10.0/website_form_builder

**Comment by simahawk** [comment]
@pedrobaeza not at all. WSF builder is to build static forms on the fly on a page.

CMS form is to make "content centric" forms, they are:
* completely untied from pages (even if you could edit a page metadata w/ it)
* everything but static (fields and behavior may vary based on implementation)
* easy to override and customize at py level and widget/xml level
* you can have create/edit/search forms (the latter can be easily used to create simple listing views too)
* reproducible (one of the most important points)

Maybe I'll stress these and other goodies in the README.
At some point I'll a probably add a way to customize them TTW (Through The Web) which in any case is not to make them work like snippets... but this is not that important ATM.

Hope this helps :)

File: cms_form/tests/fake_models.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return _fields
+
+    def _form_validate_a_float(self, value, **request_values):
+        """Specific validator for `a_float` field."""
+        value = float(value or '0')
+        return not value > 5, 'Must be greater than 5!'
+
+    def _form_validate_char(self, value, **request_values):
+        """Specific validator for all `char` fields."""
+        return not len(value) > 8, 'Text lenght must be greater than 8!'
```
Comment (ghost) [inline cms_form/tests/fake_models.py]: s/lenght/length

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**